examples_tests/22.RaytracedAO/raytraceCommon.glsl

```diff
@@ -252,14 +260,82 @@ vec3 load_normal_and_prefetch_textures(
 	return geomNormal;
 }
 
+vec3 nbl_glsl_unormSphericalToCartesian(in vec2 uv, out float sinTheta)
+{
+	vec3 dir;
+	nbl_glsl_sincos((uv.x-0.5)*2.f*nbl_glsl_PI,dir.y,dir.x);
+	nbl_glsl_sincos(uv.y*nbl_glsl_PI,sinTheta,dir.z);
+	dir.xy *= sinTheta;
+	return dir;
+}
+
+// return regularized pdf of sample
+float Envmap_regularized_deferred_pdf(in vec3 rayDirection)
+{
+	const ivec2 luminanceMapSize = textureSize(luminance[0], 0);
+	uint lastLuminanceMip = uint(log2(luminanceMapSize.x)); // TODO: later turn into push constant
+	const vec2 envmapUV = nbl_glsl_sampling_generateUVCoordFromDirection(rayDirection);
+
+	float sinTheta = length(rayDirection.zx);
+	float sumLum = texelFetch(luminance[lastLuminanceMip], ivec2(0), 0).r;
+	float lum = textureLod(luminance[0], envmapUV, 0).r;
+	float bigfactor = float(luminanceMapSize.x*luminanceMapSize.y)/sumLum;
+	return bigfactor*(lum/(sinTheta*2.0f*nbl_glsl_PI*nbl_glsl_PI));
+}
+
+void Envmap_generateRegularizedSample_and_pdf(out float pdf, out nbl_glsl_LightSample lightSample, in nbl_glsl_AnisotropicViewSurfaceInteraction interaction, in vec2 rand)
```

Comment:
call it
```glsl
vec3 nbl_glsl_sampling_envmap_HierarchicalWarp_generate(out float pdf, in vec2 rand)
```
and move it to a builtin header

then also make an override
```glsl
nbl_glsl_LightSample nbl_glsl_sampling_envmap_HierarchicalWarp_generate(out float pdf, in vec2 rand, in nbl_glsl_AnisotropicViewSurfaceInteraction interaction)
{
   return nbl_glsl_createLightSample(nbl_glsl_sampling_envmap_HierarchicalWarp_generate(pdf,rand), interaction);
}
```

Reply:
also an override which takes already provided 
```glsl

	const ivec2 warpMapSize = textureSize(warpMap,0);
	const ivec2 lastWarpMapPixel = warpMapSize - ivec2(1.f);
	float pdfConstant = 1.f/(4.f*nbl_glsl_PI*float(lastWarpMapPixel.x*lastWarpMapPixel.y));
```
would be good

Reply:
Done